google/cloud/aiplatform/metadata/schema/base_artifact.py

```diff
@@ -0,0 +1,134 @@
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        **kwargs:
+            Optional. Additional Args that will be passed directly to the Artifact base method for backward compatibility.
+    """
+
+    ARTIFACT_PROPERTY_KEY_RESOURCE_NAME = "resourceName"
+    SCHEMA_TITLE = "system.Artifact"
+
+    def __init__(
+        self,
+        schema_title: Optional[str] = None,
+        resource_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+        **kwargs,
+    ):
+
+        """Initializes the Artifact with the given name, URI and metadata."""
+        self.schema_title = BaseArtifactSchema.SCHEMA_TITLE
+        if schema_title:
+            self.schema_title = schema_title
+        self.resource_name = resource_name
+
+        self.resource_id = None
+        if resource_name:
+            # Temporary work around while Artifact.create takes resource_id instead of resource_name
+            # TODO: switch to using resouce_name only when create resouce supports it.
+            self.resource_id = resource_name.split("/")[-1]
+
+        self.uri = uri
+        self.display_name = display_name
+        self.schema_version = schema_version or constants._DEFAULT_SCHEMA_VERSION
+        self.description = description
+        self.metadata = metadata
+        self.state = state
+        self.kwargs = kwargs
+
+    def create(
+        self,
+        metadata_store_id: Optional[str] = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Creates a new Metadata Artifact.
+
+        Args:
+            metadata_store_id (str):
+                Optional. The <metadata_store_id> portion of the resource name with
+                the format:
+                projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>
+                If not provided, the MetadataStore's ID will be set to "default".
+            project (str):
+                Optional. Project used to create this Artifact. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Optional. Location used to create this Artifact. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials used to create this Artifact. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            Artifact: Instantiated representation of the managed Metadata Artifact.
+        """
+        self.artifact = artifact.Artifact.create(
```

Comment:
Why do we associate this to `self`?

```Python
base_artifact = BaseArtifactSchema(...)

artifact = base_artifact.create()
other_artifact = base_artifact.create()
```
It seems arbitrary that `base_artifact.artifact` would be `other_artifact`

Reply:
good point! removed the assign to self. this method will return the artifact only.